x/programs/v2/config.go

```diff
@@ -0,0 +1,321 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package v2
+
+import (
+	"github.com/ava-labs/avalanchego/utils/units"
+	"github.com/bytecodealliance/wasmtime-go/v14"
+)
+
+type CompileStrategy uint8
+
+const (
+	// CompileWasm will compile the wasm module before instantiating it.
+	CompileWasm CompileStrategy = iota
+	// PrecompiledWasm accepts a precompiled wasm module serialized by an Engine.
+	PrecompiledWasm
+)
+
+var (
+	DefaultMaxWasmStack         = 256 * units.MiB // 256 MiB
+	DefaultSIMD                 = false
+	DefaultEnableReferenceTypes = false
+	DefaultEnableBulkMemory     = false
+	DefaultProfilingStrategy    = wasmtime.ProfilingStrategyNone
+	DefaultMultiValue           = false
+
+	defaultWasmThreads                  = false
+	defaultFuelMetering                 = true
+	defaultWasmMultiMemory              = false
+	defaultWasmMemory64                 = false
+	defaultCompilerStrategy             = wasmtime.StrategyCranelift
+	defaultEpochInterruption            = true
+	defaultNaNCanonicalization          = "true"
+	defaultCraneliftOptLevel            = wasmtime.OptLevelSpeed
+	defaultEnableCraneliftDebugVerifier = false
+	defaultEnableDebugInfo              = false
+)
```

Comment:
Unrelated to this PR... we need to figure out a way to disable floats... The underlying engine can definitely be configured not to use floats, but not sure if that configuration option is exposed (in Rust or Go)